pkg/providers/instance/instance.go

```diff
@@ -244,7 +271,7 @@ func (p *Provider) createNetworkInterface(ctx context.Context, nicName string, l
 		return "", err
 	}
 
-	nic := p.newNetworkInterfaceForVM(nicName, backendPools, instanceType)
+	nic := p.newNetworkInterfaceForVM(ctx, nicName, backendPools, instanceType)
```

Comment:
these refactors for gocyclo for newNetworkInterfaceForVM and newVirtualMachine are kind of sloppy in the sense that we pass a lot of the same parameters down and the responsibilities are not that different. 


Should we refactor this elsewhere to combine the wrapper newNetworkInterfaceForVM with its lower level functions?